cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,189 @@
+"""Tests for cloud provider's volumes"""
+
+import pytest
+from fauxfactory import gen_alphanumeric
+from cfme.cloud.instance import Instance
+from cfme.cloud.volume import (creation_form, device_input, get_volume_name,
+                               list_tbl, select_vm, select_volume, Volume)
+from cfme.fixtures import pytest_selenium
+from cfme.web_ui import InfoBlock, toolbar
+from cfme.web_ui.flash import assert_message_contain
+from cfme.web_ui.form_buttons import (attach_volume, detach_volume, FormButton,
+                                      submit_changes)
+from random import choice
+from utils import testgen, version
+from utils.appliance.implementations.ui import navigate_to
+from utils.wait import wait_for
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack'], scope='module')
+
+
+PROD_VERSION = version.current_version().product_version()
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_create_volume(provider):
+    """Creates a volume for given cloud provider"""
+    navigate_to(Volume, 'All')
+    toolbar.select('Configuration', 'Add a new Cloud Volume')
+    vname = gen_alphanumeric()
+    vsize = 1
+    volume_data = dict(volume_name=vname,
+                       volume_size=vsize,
+                       cloud_tenant=provider.mgmt.list_tenant()[0])
+    creation_form.fill(volume_data)
+
+    pytest_selenium.click(FormButton('Add', ng_click='addClicked()'))
```

Comment:
Then it is perfectly valid, but then please move the button definition out of the test.